### PR TITLE
FIX PDO in PHP 8

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-framework/issues/10285

MySQLi PR - https://github.com/silverstripe/silverstripe-framework/pull/10281/files

One of the methods being fixed here is a better approach for what went into the `4` branch here https://github.com/silverstripe/silverstripe-framework/pull/10360/files#diff-798255e3f354dbb92e5f7a8702e95634213431cb0553f2e12f633357b7edacdaR562

### CODE REVIEW

File: src/ORM/Connect/PDOConnector.php [hunk 1]
```diff
@@ -557,8 +563,11 @@ public function transactionRollback($savepoint = null)
             }
         }
 
+        // Note: $this->inTransaction may not match the 'in-transaction' state in PDO
         $this->inTransaction = false;
-        return $this->pdoConnection->rollBack();
+        if ($this->pdoConnection->inTransaction()) {
+            $this->pdoConnection->rollBack();
```
Comment (GuySartorelli) [inline src/ORM/Connect/PDOConnector.php:569]:
```suggestion
            return $this->pdoConnection->rollBack();
```
Returns a bool: https://www.php.net/manual/en/pdo.rollback.php

File: src/ORM/Connect/PDOConnector.php [hunk 2]
```diff
@@ -557,8 +563,11 @@ public function transactionRollback($savepoint = null)
...
+        }
```
Comment (GuySartorelli) [inline src/ORM/Connect/PDOConnector.php:570]:
```suggestion
        }
        // return false because it did not rollback.
        return false;
```

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: src/ORM/Connect/PDOConnector.php (hunk 1 above)
Comment (emteknetnz) [reply src/ORM/Connect/PDOConnector.php:569]: Updated

File: src/ORM/Connect/PDOConnector.php (hunk 2 above)
Comment (emteknetnz) [reply src/ORM/Connect/PDOConnector.php:570]: Updated

**Review by GuySartorelli — APPROVED** [review]

**Pull request merged**